### DISCUSSION TITLE
Interruption occurred in the converting stage

### DISCUSSION THREAD

**Discussion by kiran-187** [vishnu-ks/Cloud-migration#189] [Q&A] [answered]
**While converting image after cloning stage the error occurs**

This issue gives the message that _**vmimport**_  role does not have enough permissions or sometimes it gives _**s3 key does not exist**_ , Even though if we have a full permissions to the user.

- Tested by giving administrator access to the user.
- Tested with the different branches and tags.
- Tested by adding additional permission to the **vmimport** role after the creation.

**Screenshots**
![Screenshot (174)](https://user-images.githubusercontent.com/126472815/221627891-942baa44-0013-48e5-b707-57432fcff6cf.png)

**xmigrate is deployed on the VM (AWS)** :
 - OS: [Ubuntu 18.04]
 - Browser [chrome]

**Comment by iamvishnuks** [comment]
Can you confirm the region of s3 bucket in aws and what region you have passed during the project creation in xmigrate?
Both region should be the same, otherwise this issue can happen.
@kiran-187

**Reply by kiran-187** [reply]
Hi @iamvishnuks , the region which we have passed during the project creation is **us-east-1** and also the bucket is created in the same region what we have passed.

**Reply by iamvishnuks** [reply]
@kiran-187 Did you check if the there's an object named `xmig-demo-slash.img` present in your s3 bucket?
Can you also paste a screenshot of the projects settings page? On the left hand side bottom u will find the settings button, mask the credentials in the screenshot before pasting it here.

**Reply by kiran-187** [reply]
There is no object in the s3 bucket @iamvishnuks                                                                                                                                                                          And the screenshot of the projects settings page is given below:                                                                                                                                             
![Screenshot (178)](https://user-images.githubusercontent.com/126472815/222052288-f5632d2b-2c3e-4edb-93b8-f415e98a0f9d.png)

**Reply by iamvishnuks** [reply]
@kiran-187 Then the issue is during the cloning phase, do you have any error message in cloning?

**Reply by kiran-187** [reply]
There is no error or warning message seen in the cloning part @iamvishnuks                                                                                                                          
![Screenshot (666)](https://user-images.githubusercontent.com/126472815/222078396-feb55164-2111-4863-8150-97831dc45810.png)

**Reply by iamvishnuks** [reply]
@kiran-187 When the migration starts it runs this playbook https://github.com/xmigrate/xmigrate/blob/main/ansible/aws/start_migration.yaml
Is there any output related to the execution of the above playbook after you hit cloning?

**Reply by kiran-187** [reply]
Nope @iamvishnuks , we did not get those output messages which are in the playbook.

**Reply by iamvishnuks** [reply]
thanks for the response @kiran-187 . I would like you to execute the below command in the source server after you hit the clone button.
`sudo ps -aux | grep start`

This will show you the command which is executed for cloning the hard disk. For some reason this command is failing to clone the disks. Hence what you could try is to execute that command in the terminal manually and see the error which is thrown.
First step is to get the command by running hitting clone button from xmigrate app and then the next step is to run the above mentioned command in the source vm to get the command which triggered the cloning process. Then the final step is to copy and paste that command in the same terminal of the source vm and wait for it to throw the error.

**Reply by kumar-cloudifyops** [reply]
<img width="746" alt="image" src="https://user-images.githubusercontent.com/83403352/224977220-b4197a6b-2bf3-495c-9208-8367706edbed.png">

@iamvishnuks  FYR<

**Reply by iamvishnuks** [reply, marked as answer]
This issue is now fixed. Please pull the latest code and try it out.
